agent/engine/dependencygraph/graph.go

```diff
@@ -287,10 +285,23 @@ func containerOrderingDependenciesCanResolve(target *apicontainer.Container,
 			// The 'target' container desires to be moved to its 'steady' state.
 			// Allow this only if the dependency container also desires to be in 'steady state'
 			return dependsOnContainerDesiredStatus == dependsOnContainer.GetSteadyStateStatus()
-
 		}
 		return false
 
+	case SuccessCondition:
+		var isDependOnContainerStoppedSuccessfully bool
```

Comment:
You should consider renaming this variable. It's hard to grok what this means exactly.